src/lib.rs

```diff
@@ -652,7 +659,13 @@ impl<T, F, P> AtomicPtr<T, F, P> {
     /// Stores `new` if the current pointer is `current`.
     ///
     /// The return value is a result indicating whether the new pointer was written and containing
-    /// the previous pointer. On success this value is guaranteed to be equal to `current`.
+    /// the previous pointer.
+    ///
+    /// # Returns
+    ///
+    /// - On success `current` is returned and `new` is guarnteed to have been stored into `self`.
+    /// - On failure the value that caused the swap to failed is returned (that value inside `self` which is
+    ///   different from `current`).
```

Comment:
I would phrase this more like the standard library docs (if not exactly like them):

> The return value is a result indicating whether the new value was written and containing the previous value. On success this value is guaranteed to be equal to `current`.